splash/qtrender_lua.py

```diff
@@ -723,6 +737,38 @@ def set_header(self, name, value):
         self.request.setRawHeader(name, value)
 
 
+@contextlib.contextmanager
+def wrapped_response(lua, reply):
+    """
+    Context manager which returns a wrapped QNetworkReply
+    suitable for using in Lua code.
+    """
+    res = _WrappedResponse(lua, reply)
+    with lua.object_allowed(res, res.attr_whitelist):
+        yield res
+
+
+class _WrappedResponse(object):
+    _attribute_whitelist = ['commands', "response"]
+
+    def __init__(self, lua, response):
+        self.lua = lua
+        self.response = response
+        self.headers = dict([(str(k), str(v)) for k, v in response.rawHeaderPairs()])
```

Comment:
Splash requires Python 2.7+, so it can be written more succinctly: 
`{str(k): str(v) for k,v in response.rawHeaderPairs()}`

While header->name mapping is convenient, it is not always correct - there could be several response headers with the same name, it is allowed by HTTP (see e.g. http://stackoverflow.com/questions/4371328/are-duplicate-http-response-headers-acceptable).